### PR TITLE
ROC-5465: Add new claim field to capture the state

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[ROC-5465](url)

### Change description ###
This PR adds state field in claim store for a claim.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/mappers/ClaimMapper.java [hunk 1]
```diff
@@ -57,10 +58,19 @@ public Claim map(int index, ResultSet result, StatementContext ctx) throws SQLEx
...
             toNullableEntity(result.getString("re_determination"), ReDetermination.class),
             toNullableLocalDateTimeFromUTC(result.getTimestamp("re_determination_requested_at")),
-            toNullableEntity(result.getString("claim_documents"), ClaimDocumentCollection.class)
+            toNullableEntity(result.getString("claim_documents"), ClaimDocumentCollection.class),
+            toNullableClaimState(result.getString("state"))
         );
     }
 
+    private ClaimState toNullableClaimState(String state) {
+        if (state == null) {
```
Comment (markry-kainos) [inline ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/mappers/ClaimMapper.java]: minor: change to ternary?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -199,6 +199,7 @@ module "claim-store-api" {
     FEATURE_TOGGLES_EMAILTOSTAFF = "${var.enable_staff_email}"
     FEATURE_TOGGLES_CCD_ENABLED = "${var.ccd_enabled}"
     FEATURE_TOGGLES_CCD_ASYNC_ENABLED = "${var.ccd_async_enabled}"
+    FEATURE_TOGGLES_SAVE_CLAIM_STATE_ENABLED = "${var.save_claim_state_enabled}"
```
Comment (markry-kainos) [inline infrastructure/main.tf]: pls add this to cmc-integration-tests and helm values.yaml

Comment (sabahirfan) [reply infrastructure/main.tf]: done

File: ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/mappers/ClaimMapper.java (hunk 1 above)
Comment (sabahirfan) [reply ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/mappers/ClaimMapper.java]: done

**Pull request merged**